### PR TITLE
Add api shield configuration GET/PUT

### PR DESCRIPTION
**Pull request by jafowler**
## Description

This PR adds support for API shield configuration GET/PUT requests. 
This is needed to help support more products under the Http Applications umbrella. 

## Has your change been tested?

I wrote some tests for both GET and PUT. Can add more if needed. 

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.https://api.cloudflare.com/#api-shield-settings-properties

### CODE REVIEW

**Comment by jafowler** [comment]
Just saw https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md, will work to update this to follow the doc.

File: api_shield.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	var asResponse APIShieldResponse
+	err = json.Unmarshal(res, &asResponse)
+	if err != nil {
+		return APIShield{}, ResultInfo{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return asResponse.Result, asResponse.ResultInfo, nil
+}
+
+func (api *API) PutAPIShieldConfiguration(ctx context.Context, rc *ResourceContainer, apiShieldData APIShield) (Response, error) {
```
Comment (jacobbednarz) [inline api_shield.go]: does `Update` make more sense here to align with the CRUD operations?

i'm also wondering if we should update the method signature to accept `params` which is it's own struct avoid tying the method signature to the potential API changes.

```suggestion
type UpdateAPIShieldParams { 
    AuthIdCharacteristics []AuthIdCharacteristics `json:"auth_id_characteristics"`
}

func (api *API) UpdateAPIShieldConfiguration(ctx context.Context, rc *ResourceContainer, params UpdateAPIShieldParams) (Response, error) {
```

File: api_shield.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+type APIShieldResponse struct {
+	Result APIShield `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+func (api *API) GetAPIShieldConfiguration(ctx context.Context, rc *ResourceContainer) (APIShield, ResultInfo, error) {
+	uri := fmt.Sprintf("/zones/%s/api_gateway/configuration?properties=auth_id_characteristics", rc.Identifier)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
```
Comment (jacobbednarz) [inline api_shield.go:40]: do you know if there are other options here? i'm leaning towards making this it's own set of parameters to ensure that we can pivot the API with breaking changes but :shrug:

Comment (jafowler) [reply api_shield.go:40]: AFICT its only the `auth_id_characteristics` https://api.cloudflare.com/#api-shield-settings-properties

File: api_shield.go (hunk 1 above)
Comment (jafowler) [reply api_shield.go]: Yeah, you're right. Lets do it.

**Comment by jacobbednarz** [comment]
thanks @jafowler - looks good!

**Comment by jafowler** [comment]
Thanks @jacobbednarz for cleaning up the tests, i'll follow that for the next few resources i'll be adding.

**Pull request merged**